packages/ui/src/components/va-modal/plugin/index.ts

```diff
@@ -36,6 +37,14 @@ const createVaModalPlugin = (app: App) => ({
       }, app?._context)
     })
   },
+  close (ok: boolean) {
+    modalInstance.props!.onClose()
```

Comment:
We need to add comments here or better explicitly add `destroy` method to modalInstance. It takes some brain power to understand why onClose is actually closing modal.